cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/overlays/BasicBlock.kt

```diff
@@ -25,15 +25,47 @@
  */
 package de.fraunhofer.aisec.cpg.graph.overlays
 
+import de.fraunhofer.aisec.cpg.PopulatedByPass
 import de.fraunhofer.aisec.cpg.graph.BranchingNode
 import de.fraunhofer.aisec.cpg.graph.Node
 import de.fraunhofer.aisec.cpg.graph.OverlayNode
+import de.fraunhofer.aisec.cpg.graph.edges.Edge
+import de.fraunhofer.aisec.cpg.graph.edges.collections.EdgeList
+import de.fraunhofer.aisec.cpg.graph.edges.collections.MirroredEdgeCollection
+import de.fraunhofer.aisec.cpg.graph.edges.unwrapping
 import de.fraunhofer.aisec.cpg.graph.statements.LoopStatement
 import de.fraunhofer.aisec.cpg.graph.statements.expressions.ComprehensionExpression
+import de.fraunhofer.aisec.cpg.passes.BasicBlockCollectorPass
 import de.fraunhofer.aisec.cpg.sarif.PhysicalLocation
 import de.fraunhofer.aisec.cpg.sarif.Region
 import java.net.URI
 import java.util.Objects
+import kotlin.reflect.KProperty
+import org.neo4j.ogm.annotation.Relationship
+import org.neo4j.ogm.annotation.RelationshipEntity
+
+@RelationshipEntity
+class BasicBlockEdge(start: Node, end: Node) : Edge<Node>(start, end) {
```

Comment:
Needs documentation, also this should be in the `.edges` package. What exactly does this edge represent? that its part of the a certain basic block? Or is it the flow between them - in which case it would be a "traditional" CFG? This edge confuses me :)

Reply:
It represents that a node belongs to a certain basic block. The CFG is represented by EOG edges between the basic blocks.